src/dbnode/storage/index/read_through_segment.go

```diff
@@ -155,6 +157,48 @@ func (r *ReadThroughSegment) Size() int64 {
 	return r.segment.Size()
 }
 
+func (r *ReadThroughSegment) SegmentData(ctx context.Context) (fst.SegmentData, error) {
+	return r.segment.SegmentData(ctx)
+}
+
+func (r *ReadThroughSegment) Freeze() {
+	r.segment.Freeze()
+}
+
+func (r *ReadThroughSegment) State() (fst.IndexSegmentState, error) {
+	return r.segment.State()
+}
+
+func (s *ReadThroughSegment) Doc(id postings.ID) (doc.Document, error) {
+	return s.Doc(id)
+}
+
+func (s *ReadThroughSegment) Docs(pl postings.List) (doc.Iterator, error) {
```

Comment:
Think a lot of these methods aren't required anymore, see changes to prev FST segment type here:
https://github.com/m3db/m3/commit/abfa755798d1ef848c09c4f05076b47941ffab23